whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy

```diff
@@ -242,29 +242,11 @@ class PostgreSQLComponent implements Storage {
                    "OR data->'@graph' @> ?"
 
         GET_SYSTEMID_BY_IRI = "SELECT id FROM $idTableName WHERE iri = ?"
+        GET_THING_MAIN_IRI_BY_SYSTEMID = "SELECT iri FROM $idTableName WHERE graphindex = 1 and mainid is true and id = ?"
         GET_DOCUMENT_BY_IRI = "SELECT lddb.id,lddb.data,lddb.created,lddb.modified,lddb.deleted FROM lddb INNER JOIN lddb__identifiers ON lddb.id = lddb__identifiers.id WHERE lddb__identifiers.iri = ?"
 
         GET_LEGACY_PROFILE = "SELECT profile FROM $profilesTableName WHERE library_id = ?"
 
-        NESTED_RELATIONS = "WITH RECURSIVE nested_relation AS (" +
-                "    SELECT id " +
-                "    FROM $dependenciesTableName " +
-                "    WHERE dependsonid = ? " +
-                "    AND relation = ANY (?) " +
-                "  UNION " +
-                "    SELECT d.id " +
-                "    FROM $dependenciesTableName d " +
-                "    INNER JOIN nested_relation n ON d.dependsonid = n.id " +
-                "    WHERE relation = ANY (?) " +
-                ") " +
-                "SELECT i.iri " +
-                "FROM nested_relation n, $idTableName i " +
-                "WHERE n.id = i.id " +
-                "AND i.graphindex = 1 " +
-                "AND i.mainid IS TRUE"
-     }
```

Comment:
Unfortunately this can be really slow sometimes.
eg `IRI = 'https://id.kb.se/term/saogf/Sk%C3%B6nlitteratur'`
`relations = ['broader', 'broadMatch', 'exactMatch']` 

Probably because of cycles in the dependency graph that don't get culled.
Maybe it is possible to fix.